src/test/java/org/jitsi/meet/test/base/FailureListener.java

```diff
@@ -170,6 +179,30 @@ public void onTestFailure(ITestResult testResult)
         }
     }
 
+    @Override
+    public void onConfigurationSuccess(ITestResult itr)
+    {}
+
+    @Override
+    public void onConfigurationFailure(ITestResult itr)
+    {
+        Object testInstance = itr.getInstance();
+        if (testInstance instanceof AbstractBaseTest)
+        {
+            // record whatever we can
+            onTestFailure(itr);
+
+            // make sure if setup fails we will cleanup
+            // when configure method @BeforeClass fails @AfterClass is not
+            // executed
+            ((AbstractBaseTest)testInstance).cleanupClass();
```

Comment:
I understand we need to take the screenshots, logs, etc, but doing the cleanup should be part of AbstractBaseTest.

Reply:
or is it that if we cleanup, there will be nothing to take the screenshots from ?

Reply:
That's correct, we need to do it before cleaning